### ISSUE TITLE
browser.execute doesn't refetch element

### ISSUE THREAD

**Issue by mgrybyk** [webdriverio/webdriverio#4801]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.16.5

**Describe the bug**

**To Reproduce**
```
browser.url("http://guinea-pig.webdriver.io")
const el = $('.box')
browser.refresh()
browser.execute(el => el, el)
```

**Expected behavior**
`browser.execute` should refetch element 

**Log**
`stale element reference: element is not attached to the page document
  (Session info: headless chrome=78.0.3904.108)`

**Additional context**
Using `el.isExisting()` doesn't help neither because it's not updating element id to the new one (and I'm not sure if t actually should)

**Comment by christian-bromann** [comment]
This is because we don't apply our middleware for browser commands. I would write a middleware for the browser object to handle this special case.

**Comment by mgrybyk** [comment]
I now have doubts about it. Using browser commands is the only way now to understand if element was refetched or not. After implementing this it would be not possible

**Comment by christian-bromann** [comment]
> Using browser commands is the only way now to understand if element was refetched or not.

What do you mean by that?

**Comment by mgrybyk** [comment]
I mean that currently the command is failing because element is stale.
This is a way to understand that element has changed.
If we fix the issue by adding middleware that would silently refetch the element how can I identify if element has changed or not?

Forget it, it should be possible by comparing element id

**Comment by vgrigoruk** [comment]
Also reproducible for `staleElement.scrollIntoView()`, as it uses `browser.execute` under the hood :(

**Comment by erwinheitzman** [comment]
In v9 this should be fixed with the introduction of the Element.execute method which automatically waits for the existence of the element before running it

**Comment by christian-bromann** [comment]
> the Element.execute method which automatically waits for the existence of the element before running it

That is not true. I raised https://github.com/webdriverio/webdriverio/pull/14270 to change that.
